test-gen-plugin/src/main/java/com/dylibso/chicory/maven/Wast2JsonWrapper.java

```diff
@@ -228,4 +231,36 @@ public String wabtArchitectureName() {
             throw new IllegalArgumentException("Detected OS is not supported: " + osName);
         }
     }
+
+    /**
+     * Compares the available wast2json version with the minimum required version
+     * using semantic version numbers. Assumes that the version is a 3 part semantic
+     * version number.
+     * @param actual the version of wast2json already installed
+     * @param required the minimum version of wast2json that's required
+     * @return true if the actual version meets the requirements, false otherwise
+     */
+    private boolean versionMatches(String actual, String required) {
+        var actualComponents = splitVersion(actual);
+        var requiredComponents = splitVersion(required);
+
+        if (actualComponents.length != 3 || requiredComponents.length != 3) {
```

Comment:
Have you tried to compile locally wabt and check the version number?
I'm afraid of causing more trouble if, in the future, we'd need to compile locally for any reason and obtain a version with something like `+<commit sha>` ... but maybe it's already fine.

Reply:
That's a good point - pushing a small modification that tries to do an exact match before falling back to semantic version checking. If either fail, it defaults to downloading the required version.

Reply:
Make sense to me, do you mind adding a couple of log messages so that is clear what its going on?